src/main/java/com/salesforce/functions/jvm/sdk/bulk/BulkApi.java

```diff
@@ -62,22 +60,21 @@ public interface BulkApi {
    *     href="https://developer.salesforce.com/docs/atlas.en-us.api_asynch.meta/api_asynch/get_job_failed_results.htm">Bulk
    *     API 2.0 and Bulk API Developer Guide</a>
    * @param jobId The ID of the job to get failed record results for.
-   * @return A CSV file with all failed records.
+   * @return An iterable containing the records that failed to be processed.
    * @throws BulkApiException If an error occurred while querying failed record results.
    */
   @SuppressWarnings("unused")
-  File getJobFailedRecordResults(String jobId) throws BulkApiException;
+  Iterable<Record> getJobFailedRecordResults(String jobId) throws BulkApiException;
 
   /**
    * Uploads data for a job using CSV data you provide.
    *
    * @param jobId The ID for the job to upload data for.
-   * @param inputStream The CSV data as a stream to upload to the job.
+   * @param records The CSV data as a stream to upload to the job.
    * @throws BulkApiException If an error occurred uploading the data.
-   * @throws IOException If an error occurred reading data from the provided stream.
    */
   @SuppressWarnings("unused")
-  void uploadJobData(String jobId, InputStream inputStream) throws BulkApiException, IOException;
+  void uploadJobData(String jobId, Iterable<Record> records) throws BulkApiException;
```

Comment:
Uses `Iterable<Record>` so that standard Java collections like `List<Record>` can be supplied or, in the case of larger data sets, an `Iterable` that provides a custom `Iterator`